### PR TITLE
[SPARK-40147][PYTHON][SQL] Make pyspark.sql.session examples self-contained

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR proposes to improve the examples in `pyspark.sql.session` by making each example self-contained with a brief explanation and a bit more realistic example.

### Why are the changes needed?

To make the documentation more readable and able to copy and paste directly in PySpark shell.

### Does this PR introduce _any_ user-facing change?

Yes, it changes the documentation

### How was this patch tested?

Manually ran each doctests. CI also runs this.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @Yikun @itholic @viirya @xinrong-meng @ueshin @zhengruifeng PTAL when you find some time.

File: python/pyspark/sql/session.py [hunk 1]
```diff
@@ -99,8 +99,15 @@ def toDF(self, schema=None, sampleRatio=None):
 
         Examples
         --------
-        >>> rdd.toDF().collect()
-        [Row(name='Alice', age=1)]
+        >>> rdd = spark.range(1).rdd.map(lambda x: tuple(x))
+        >>> rdd.collect()
```
Comment (viirya) [inline python/pyspark/sql/session.py:103]: Hm? I don't see `toDF()` in this example.

File: python/pyspark/sql/session.py [hunk 2]
```diff
@@ -1185,6 +1363,22 @@ def readStream(self) -> DataStreamReader:
...
         -------
         :class:`DataStreamReader`
+
+        Examples
+        --------
+        >>> spark.readStream
+        <pyspark.sql.streaming.readwriter.DataStreamReader object ...>
+
+        The example below uses Rate source that generates rows continously.
+        After that, we operate a modulo by 3, and then writes the stream out to the console.
```
Comment (viirya) [inline python/pyspark/sql/session.py:1373]:
```suggestion
        After that, we operate a modulo by 3, and then write the stream out to the console.
```

**Review by viirya — APPROVED** [review]

**Comment by viirya** [comment]
Two minor comments

File: python/pyspark/sql/session.py [hunk 3]
```diff
@@ -99,8 +99,15 @@ def toDF(self, schema=None, sampleRatio=None):
 
         Examples
         --------
-        >>> rdd.toDF().collect()
-        [Row(name='Alice', age=1)]
+        >>> rdd = spark.range(1).rdd.map(lambda x: tuple(x))
+        >>> rdd.collect()
+        [(0,)]
+        >>> spark.range(1).show()
```
Comment (HyukjinKwon) [inline python/pyspark/sql/session.py:105]:
```suggestion
        >>> rdd.toDF().show()
```

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**